library/Fission/User/Username/Types.hs

```diff
@@ -0,0 +1,45 @@
+module Fission.User.Username.Types (Username (..)) where
+
+import Fission.Prelude
+
+import Data.Swagger
+import Servant
+
+import qualified RIO.ByteString.Lazy as Lazy
+import qualified Network.IPFS.Internal.UTF8 as UTF8
+
+newtype Username = Username { username :: Text }
+  deriving          ( Eq
+                    , Generic
+                    , Show
+                    )
```

Comment:
You should push these into one of the explicit deriving strategies below. Probably anyclass.

Reply:
Not really sure the difference between these :thinking: definitely something I need to look into
The best way to get the compiler to stop complaining seemed to be
```haskell
newtype Username = Username { username :: Text }
  deriving          ( Generic )
  deriving anyclass ( ToParamSchema )
  deriving newtype  ( Show
                    , Eq
                    , IsString
                    , ToHttpApiData
                    )
```